### PR TITLE
Feature like dislike

### LINKED ISSUE #6: Like Dislike feature
I would love to work on the like/dislike feature for the application.

The idea is to add a separate model for liked items, where the schema contains recipeId, userId, and a timeStamp. When the "Get Recipe" API is called, it will run a database query to return the like count. Additionally, under the user section, it will return the list of items liked by the user.

Please assign me this feature under GSSoC Extended.

### PR DESCRIPTION
**Pull request by sadath2001**
## What does this PR do?

Likes, disliked recipies, shows the liked recipies in the user profile, also syncs the recipie like/unlike accross all the userts
Formatted the code to look standard.

Fixes #6 

## How should this be tested?

- Like a recipe
- check the liked recipies in user profile
- login into another account to see if the liked count has increased

### Required

- [X] Filled out the "How to test" section in this PR
- [X] Read Contributing guide
- [X] Self-reviewed my own code
- [X] Checked for warnings, there are none
- [X] Removed all `console.logs`
- [X] The changes don't cause any responsiveness issues

### CODE REVIEW

**Comment by sadath2001** [comment]
ill resolve these conflicts

File: frontend/src/Components/Cards.jsx [hunk 1]
```diff
@@ -1,81 +1,143 @@
...
+            },
           }
-        })
+        );
         if (updateShareCount.status === 200) {
-          const postURL = window.location.origin + '/recipe/' + dish._id
-          // console.log(window.location.origin)
+          const postURL = window.location.origin + "/recipe/" + dish._id;
+          console.log(window.location.origin);
           const updatedRecipes = [...recipes];
```
Comment (AlfiyaSiddique) [inline frontend/src/Components/Cards.jsx:63]: can remove this @sadath2001

File: frontend/src/Components/Cards.jsx [hunk 2]
```diff
@@ -1,81 +1,143 @@
...
-      <div className="border-2 border-gray-200 border-opacity-60 rounded-lg overflow-hidden" onClick={handleClick}>
+    <div className="p-4">
+      <div className="border-2 border-gray-200 border-opacity-60 rounded-lg overflow-hidden">
         <img
-          className="lg:h-48 md:h-36 w-full object-cover object-center"
+          className="lg:h-48 md:h-36 w-full object-cover object-center cursor-pointer"
           src={dish.image}
           alt={dish.name}
-          loading="lazy"
+          onClick={handleClick}
```
Comment (AlfiyaSiddique) [inline frontend/src/Components/Cards.jsx:140]: why did you remove lazy loading?

**Comment by AlfiyaSiddique** [comment]
@sadath2001 some comments rest looks goo to me

**Review by sadath2001 — COMMENTED** [review]
FIxed the changes as suggested

**Comment by sadath2001** [comment]
hi @AlfiyaSiddique , made the changes

**Review by AlfiyaSiddique — APPROVED** [review]
